File: bubble_sort_simple.cpp
/* bubble_sort.cpp*/

/*******************************
A program providing a non OOP
solution to the bubble sort.
Sorts lists stored in arrays
in ascending order.
Author: Samuel Patrick
Date: 9/10/16
********************************/
#include <iostream>
using namespace std;

void main()
{
	//Array Size for user input
	const int SIZE = 5;

	//Counter to count the number of swaps
	int swaps = 0;
	//Marker for the position in the list
	int position = 0;
	//Temporary variable for swaps
	int tempN;
	char tempC;
	//Variable for user input
	char input = 'y';
	//Flag to tell if a list is sorted
	bool sorted = false;
	//Array for user input set to -1
	int user[SIZE] = { -1 };
	//Provided arrays to sort
	int testA[] = { 1, 3, 4, 7, 2, 5 };
	int testB[] = { 1, 2, 3, 4, 5 };
	int testC[] = { 5, 4, 3, 2, 1 };
	int testD[] = { 10, 4, -6, 2, 15, -2 };
	char testE[] = { 'b', 'i', 'n', 'g', 'o' };

	//loop that keeps going over the list until it sorted
	//we cheat because we know in advance the size of the array we are sorting
	while (!sorted)
	{
		position = 0;
		sorted = true;

		//loop through the list from start to end
		//we stop the loop at size-1 since we are comparing n and n+1 elements
		while (position < 5)
		{
			if (testA[position] > testA[position + 1])
			{
				tempN = testA[position];
				testA[position] = testA[position + 1];
				testA[position + 1] = tempN;
				swaps = swaps + 1;
				sorted = false;
			}
			position = position + 1;
		}
	}

	//print the array contents
	position = 0;
	cout << "List A: ";
	//we again cheat since we know the size of the list
	while (position < 6)
	{
		cout << testA[position] << " ";
		position = position + 1;
	}
	//set up for next list
	cout << "\n Number of swaps: " << swaps << endl;
	sorted = false;
	swaps = 0;

	//loop that keeps going over the list until it sorted
	//we cheat because we know in advance the size of the array we are sorting
	while (!sorted)
	{
		position = 0;
		sorted = true;

		//loop through the list from start to end
		//we stop the loop at size-1 since we are comparing n and n+1 elements
		while (position < 4)
		{
			if (testB[position] > testB[position + 1])
			{
				tempN = testB[position];
				testB[position] = testB[position + 1];
				testB[position + 1] = tempN;
				swaps = swaps + 1;
				sorted = false;
			}
			position = position + 1;
		}
	}

	//print the array contents
	position = 0;
	cout << "List B: ";
	//we again cheat since we know the size of the list
	while (position < 5)
	{
		cout << testB[position] << " ";
		position = position + 1;
	}
	//set up for next list
	cout << "\n Number of swaps: " << swaps << endl;
	sorted = false;
	swaps = 0;

	//loop that keeps going over the list until it sorted
	//we cheat because we know in advance the size of the array we are sorting
	while (!sorted)
	{
		position = 0;
		sorted = true;

		//loop through the list from start to end
		//we stop the loop at size-1 since we are comparing n and n+1 elements
		while (position < 4)
		{
			if (testC[position] > testC[position + 1])
			{
				tempN = testC[position];
				testC[position] = testC[position + 1];
				testC[position + 1] = tempN;
				swaps = swaps + 1;
				sorted = false;
			}
			position = position + 1;
		}
	}

	//print the array contents
	position = 0;
	cout << "List C: ";
	//we again cheat since we know the size of the list
	while (position < 5)
	{
		cout << testC[position] << " ";
		position = position + 1;
	}
	//set up for next list
	cout << "\n Number of swaps: " << swaps << endl;
	sorted = false;
	swaps = 0;

	//loop that keeps going over the list until it sorted
	//we cheat because we know in advance the size of the array we are sorting
	while (!sorted)
	{
		position = 0;
		sorted = true;

		//loop through the list from start to end
		//we stop the loop at size-1 since we are comparing n and n+1 elements
		while (position < 5)
		{
			if (testD[position] > testD[position + 1])
			{
				tempN = testD[position];
				testD[position] = testD[position + 1];
				testD[position + 1] = tempN;
				swaps = swaps + 1;
				sorted = false;
			}
			position = position + 1;
		}
	}

	//print the array contents
	position = 0;
	cout << "List D: ";
	//we again cheat since we know the size of the list
	while (position < 6)
	{
		cout << testD[position] << " ";
		position = position + 1;
	}
	//set up for next list
	cout << "\n Number of swaps: " << swaps << endl;
	sorted = false;
	swaps = 0;

	//loop that keeps going over the list until it sorted
	//we cheat because we know in advance the size of the array we are sorting
	while (!sorted)
	{
		position = 0;
		sorted = true;

		//loop through the list from start to end
		//we stop the loop at size-1 since we are comparing n and n+1 elements
		while (position < 4)
		{
			if (testE[position] > testE[position + 1])
			{
				tempC = testE[position];
				testE[position] = testE[position + 1];
				testE[position + 1] = tempC;
				swaps = swaps + 1;
				sorted = false;
			}
			position = position + 1;
		}
	}

	//print the array contents
	position = 0;
	cout << "List E: ";
	//we again cheat since we know the size of the list
	while (position < 5)
	{
		cout << testE[position] << " ";
		position = position + 1;
	}
	//set up for next list
	cout << "\n Number of swaps: " << swaps << endl;
	sorted = false;
	swaps = 0;


	//A loop to get a list from the user and sort it
	while (input == 'Y' || input == 'y')
	{
		//Get the list from the user
		cout << "\nPlease input the first number: ";
		cin >> user[0];
		cout << "Please input the second number: ";
		cin >> user[1];
		cout << "Please input the third number: ";
		cin >> user[2];
		cout << "Please input the fourth number: ";
		cin >> user[3];
		cout << "Please input the fifth number: ";
		cin >> user[4];

		//loop that keeps going over the list until it sorted
		//we cheat because we know in advance the size of the array we are sorting
		while (!sorted)
		{
			position = 0;
			sorted = true;

			//loop through the list from start to end
			//we stop the loop at size-1 since we are comparing n and n+1 elements
			while (position < SIZE - 2)
			{
				if (user[position] > user[position + 1])
				{
					tempN = user[position];
					user[position] = user[position + 1];
					user[position + 1] = tempN;
					swaps = swaps + 1;
					sorted = false;
				}
				position = position + 1;
			}
		}

		//print the array contents
		position = 0;
		cout << "Sorted User List: ";
		//we again cheat since we know the size of the list
		while (position < SIZE - 1)
		{
			cout << user[position] << " ";
			position = position + 1;
		}
		//set up for next list
		cout << "\n Number of swaps: " << swaps << endl;
		sorted = false;
		swaps = 0;

		//Check if the user wants to enter another list
		cout << "Enter another list? (Y/N): ";
		cin >> input;
	}

};

File: bubble_sort_simple.py
#!C:\Python34\python.exe

## bubble_sort.py
## A program providing a non OOP solution to the bubble sort.
## Sorts lists stored in arrays in ascending order.
## Author: Samuel Patrick
## Date: 9/13/16

#Counter to track the swaps
swaps = 0
#Marker for the position in the list
position = 0
#Variable for swaps
temp = 0
#User input
user_in = "y"
#Flag to tell if the list is sorted
sorted = False

#lists to be sorted
testA = [1,3,4,7,2,5]
testB = [1,2,3,4,5]
testC = [5,4,3,2,1]
testD = [10,4,-6,2,15,-2]
testE = ['b','i','n','g','o']
user = [0,0,0,0,0]

while sorted == False:
	position = 0
	sorted = True
	
	#compare all the elements in the list
	while position < 5:
		if testA[position] > testA[position+1]:
			temp = testA[position]
			testA[position] = testA[position+1]
			testA[position+1] = temp
			swaps = swaps + 1
			sorted = False
		
		position = position + 1

print ("List A:", end=" ")		
#python cheats when it comes to lists
for each in testA:
	print (each, end=" ")

print ()
print ("swaps =", swaps, sep=" ")
swaps = 0
sorted = False

while sorted == False:
	position = 0
	sorted = True
	
	#compare all the elements in the list
	while position < 4:
		if testB[position] > testB[position+1]:
			temp = testB[position]
			testB[position] = testB[position+1]
			testB[position+1] = temp
			swaps = swaps + 1
			sorted = False
		
		position = position + 1

print ("List B:", end=" ")
#python cheats when it comes to lists		
for each in testB:
	print (each, end=" ")

print ()
print ("swaps =", swaps, sep=" ")
swaps = 0
sorted = False

while sorted == False:
	position = 0
	sorted = True
	
	#compare all the elements in the list
	while position < 4:
		if testC[position] > testC[position+1]:
			temp = testC[position]
			testC[position] = testC[position+1]
			testC[position+1] = temp
			swaps = swaps + 1
			sorted = False
		
		position = position + 1

print ("List C:", end=" ")
#python cheats when it comes to lists		
for each in testC:
	print (each, end=" ")

print ()
print ("swaps =", swaps, sep=" ")
swaps = 0
sorted = False

while sorted == False:
	position = 0
	sorted = True
	
	#compare all the elements in the list
	while position < 5:
		if testD[position] > testD[position+1]:
			temp = testD[position]
			testD[position] = testD[position+1]
			testD[position+1] = temp
			swaps = swaps + 1
			sorted = False
		
		position = position + 1

print ("List D:", end=" ")
#python cheats when it comes to lists		
for each in testD:
	print (each, end=" ")

print ()
print ("swaps =", swaps, sep=" ")
swaps = 0
sorted = False

while sorted == False:
	position = 0
	sorted = True
	
	#compare all the elements in the list
	while position < 4:
		if testE[position] > testE[position+1]:
			temp = testE[position]
			testE[position] = testE[position+1]
			testE[position+1] = temp
			swaps = swaps + 1
			sorted = False
		
		position = position + 1

print ("List E:", end=" ")
#python cheats when it comes to lists		
for each in testE:
	print (each, end=" ")

print ()
print ("swaps =", swaps, sep=" ")
swaps = 0
sorted = False

#loop to get user input and sort it
while user_in == 'y' or user_in == 'Y':
	user[0] = input('Please input the first number: ')
	user[1] = input('Please input the second number: ')
	user[2] = input('Please input the third number: ')
	user[3] = input('Please input the fourth number: ')
	user[4] = input('Please input the fifth number: ')
	
	while sorted == False:
		position = 0
		sorted = True
	
		#compare all the elements in the list
		while position < 4:
			if user[position] > user[position+1]:
				temp = user[position]
				user[position] = user[position+1]
				user[position+1] = temp
				swaps = swaps + 1
				sorted = False
		
			position = position + 1

	print ("User List:", end=" ")
	#python cheats when it comes to lists		
	for each in user:
		print (each, end=" ")

	print ()
	print ("swaps =", swaps, sep=" ")
	swaps = 0
	sorted = False
	
	user_in = input('Enter another list? Y/N: ')
